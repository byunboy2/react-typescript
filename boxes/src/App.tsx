import BoxList from "./BoxList";

/**
 *  Renders the App
 *
 *  Props: none
 *
 *  States: none
 *
 *  App --> BoxList
 */
function App() {
  return (
    <div>
      <BoxList />
    </div>
  );
}

export default App;
