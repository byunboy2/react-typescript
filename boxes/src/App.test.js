import { render } from "@testing-library/react";
import App from "./App";

describe("Tests the App component", function () {

  it("renders without crashing", function () {
    render(<App />);
  });

  it("matches the snapshot", function () {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
