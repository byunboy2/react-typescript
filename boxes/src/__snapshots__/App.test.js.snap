// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the App component matches the snapshot 1`] = `
<div>
  <div>
    <div>
      <div>
        <form>
          <div>
            <label
              for="newBox-height"
            >
              Height
            </label>
            <input
              id="newBox-height"
              name="height"
              value=""
            />
          </div>
          <div>
            <label
              for="newBox-width"
            >
              Width
            </label>
            <input
              id="newBox-width"
              name="width"
              value=""
            />
          </div>
          <div>
            <label
              for="newBox-backgroundColor"
            >
              Background Color
            </label>
            <input
              id="newBox-backgroundColor"
              name="backgroundColor"
              value=""
            />
          </div>
          <button
            class="NewBoxForm-addBtn"
            id="NewBoxForm-addBtn"
          >
            Add a new box!
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`matches the snapshot 1`] = `
<div>
  <div>
    <div>
      <div>
        <form>
          <div>
            <label
              for="newBox-height"
            >
              Height
            </label>
            <input
              id="newBox-height"
              name="height"
              value=""
            />
          </div>
          <div>
            <label
              for="newBox-width"
            >
              Width
            </label>
            <input
              id="newBox-width"
              name="width"
              value=""
            />
          </div>
          <div>
            <label
              for="newBox-backgroundColor"
            >
              Background Color
            </label>
            <input
              id="newBox-backgroundColor"
              name="backgroundColor"
              value=""
            />
          </div>
          <button
            class="NewBoxForm-addBtn"
          >
            Add a new box!
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
